src/TestProblem/Hydro/Grackle_Comoving/Aux_CoolingCurve.cpp

```diff
@@ -0,0 +1,206 @@
+      }
+      double mu                    = my_temperature[0] / (my_fields.internal_energy[0] * (my_gamma[0] - 1.) * temperature_units);
+      my_fields.internal_energy[0] = pow(10.0, logT) / (mu * (my_gamma[0] - 1.) * temperature_units);
+
+      if (calculate_temperature(&my_units, &my_fields, my_temperature) == 0) {
+        Aux_Error( ERROR_INFO, "Error in calculate_temperature.\n");
+      }
+
+      // check convergence
+      double diff = 0.0;
+      if ( GRACKLE_PRIMORDIAL >= GRACKLE_PRI_CHE_NSPE6 ) {
+      diff = fmax(diff, fabs(my_fields.HI_density   [0] - buf_HI_density   ) / my_fields.HI_density   [0]);
+      diff = fmax(diff, fabs(my_fields.HII_density  [0] - buf_HII_density  ) / my_fields.HII_density  [0]);
+      diff = fmax(diff, fabs(my_fields.HeI_density  [0] - buf_HeI_density  ) / my_fields.HeI_density  [0]);
+      diff = fmax(diff, fabs(my_fields.HeII_density [0] - buf_HeII_density ) / my_fields.HeII_density [0]);
+      diff = fmax(diff, fabs(my_fields.HeIII_density[0] - buf_HeIII_density) / my_fields.HeIII_density[0]);
+      diff = fmax(diff, fabs(my_fields.e_density    [0] - buf_e_density    ) / my_fields.e_density    [0]);
+      }
+      if ( GRACKLE_PRIMORDIAL >= GRACKLE_PRI_CHE_NSPE9 ) {
+      diff = fmax(diff, fabs(my_fields.HM_density   [0] - buf_HM_density   ) / my_fields.HM_density   [0]);
+      diff = fmax(diff, fabs(my_fields.H2I_density  [0] - buf_H2I_density  ) / my_fields.H2I_density  [0]);
+      diff = fmax(diff, fabs(my_fields.H2II_density [0] - buf_H2II_density ) / my_fields.H2II_density [0]);
+      }
+      if ( GRACKLE_PRIMORDIAL >= GRACKLE_PRI_CHE_NSPE12 ) {
+      diff = fmax(diff, fabs(my_fields.DI_density   [0] - buf_DI_density   ) / my_fields.DI_density   [0]);
+      diff = fmax(diff, fabs(my_fields.DII_density  [0] - buf_DII_density  ) / my_fields.DII_density  [0]);
+      diff = fmax(diff, fabs(my_fields.HDI_density  [0] - buf_HDI_density  ) / my_fields.HDI_density  [0]);
+      }
+      diff = fmax(diff, fabs(my_temperature         [0] - pow(10.0, logT)  ) / my_temperature         [0]);
+
+      if (diff < 1.e-3)   break; // converged
+
+      iter++;
+    }
+    maxiter = iter > maxiter ? iter : maxiter;
+
+    // Calculate cooling time.
+    if (calculate_cooling_time(&my_units, &my_fields, my_cooling_time) == 0) {
+      Aux_Error( ERROR_INFO, "Error in calculate_cooling_time.\n");
+    }
+
+    double mu    = my_temperature[0] / (my_fields.internal_energy[0] * (my_gamma[0] - 1.) * temperature_units);
+    double nden  = my_fields.density[0] / mu / Const_mp * my_units.density_units;
+    double eden  = my_fields.density[0] * my_fields.internal_energy[0] * my_units.density_units * my_units.velocity_units * my_units.velocity_units;
+    double lcool = eden / fabs(my_cooling_time[0] * my_units.time_units) / nden / nden;
+
+    fprintf(fp, " %30.8e%30.8e\n", my_temperature[0], lcool);
+  }
+
+  fclose(fp);
+
+  Aux_Message( stdout, "  done %s (maxiter = %d)\n", ofname, maxiter );
```

Comment:
I guess the word `done` should be put in the Line. 380 of `Init_TestProb_Grackle_Comoving.cpp`?